AWS Docs (#446)
edgelesssys/constellation: docs/docs/getting-started/install.md

Follow Google's guide on [understanding](https://cloud.google.com/iam/docs/understanding-roles) and [assigning roles](https://cloud.google.com/iam/docs/granting-changing-revoking-access).

</tabItem>
<tabItem value="aws" label="AWS">

> In the following steps you will use Constellation to perform two tasks: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure. We therefore provide a set of permissions for each.

Reviewer: ```suggestion
To set up a Constellation cluster, you need to perform two tasks that require permissions: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure.
```

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "sts:GetCallerIdentity",
                "ec2:DescribeAccountAttributes",
                "iam:CreateRole",
                "iam:CreatePolicy",
                "iam:GetPolicy",
                "iam:GetRole",
                "iam:GetPolicyVersion",
                "iam:ListRolePolicies",
                "iam:ListAttachedRolePolicies",
                "iam:CreateInstanceProfile",
                "iam:AttachRolePolicy",
                "iam:GetInstanceProfile",
                "iam:AddRoleToInstanceProfile",
                "iam:PassRole",
                "iam:RemoveRoleFromInstanceProfile",
                "iam:DetachRolePolicy",
                "iam:DeleteInstanceProfile",
                "iam:ListPolicyVersions",
                "iam:ListInstanceProfilesForRole",
                "iam:DeletePolicy",
                "iam:DeleteRole"
            ],
            "Resource": "*"
        }
    ]
}
```

To create the infrastructure, you can either use a predefined role from Amazon,

> such as `AmazonEC2FullAccess`, or use the following minimal set of permissions:

Reviewer 2: `AmazonEC2FullAccess` alone is not sufficient since i.e. `logs:*` permissions form below are missing.

Follow Google's guide on [understanding](https://cloud.google.com/iam/docs/understanding-roles) and [assigning roles](https://cloud.google.com/iam/docs/granting-changing-revoking-access).

</tabItem>
<tabItem value="aws" label="AWS">

> In the following steps you will use Constellation to perform two tasks: create policies & infrastructure.

Author: I added more context to help with the motivation, @thomasten.

                "elasticloadbalancing:AddTags",
                "elasticloadbalancing:DescribeLoadBalancerAttributes",
                "elasticloadbalancing:CreateListener",
                "elasticloadbalancing:DescribeListeners",
                "logs:DeleteLogGroup",
                "elasticloadbalancing:DeleteListener",
                "ec2:DisassociateRouteTable",
                "autoscaling:UpdateAutoScalingGroup",
                "elasticloadbalancing:DeleteLoadBalancer",
                "autoscaling:SetInstanceProtection",
                "ec2:DescribeNetworkInterfaces",
                "ec2:DeleteRouteTable",
                "ec2:DeleteNatGateway",
                "ec2:DetachInternetGateway",
                "ec2:DisassociateAddress",
                "ec2:ReleaseAddress",
                "ec2:DeleteInternetGateway",
                "ec2:DeleteSubnet",
                "autoscaling:DeleteAutoScalingGroup",
                "ec2:DeleteLaunchTemplate",
                "elasticloadbalancing:DeleteTargetGroup",
                "ec2:DeleteSecurityGroup",
                "ec2:DeleteVpc"
            ],
            "Resource": "*"
        }
    ]
}
```

> Follow Amazons' guide on [understanding](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) and [managing policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html), and [roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_iam-tags.html).

Reviewer 3: In my opinion the whole sentence is a bit awkward to read.
"Follow Amazon's guide on understanding and managing policies" by itself sounds fine, though I think both articles are actually about understanding the purpose of certain kinds of policies.

"Follow Amazon's guide on understanding and managing policies_, and roles_" sounds off to me. The comma implies a break in speech, yet "and roles" is meant to be read as "Follow Amazon's guide on understanding [...] roles" or "Follow Amazon's guide on managing [...] roles".

Not sure which one, though. And having two "ands" in the sentence is also not good, which makes it a confusing enumeration.

I would rewrite this, though I don't have a concrete suggestion as I don't fully know the articles and context of the linked articles to make a suggestion.

Author: I agree! I have removed the last link since it was off-context. The first two resources should provide enough information to know how policies work.
